File: app/src/main/java/com/autoscout24/carfinder/ui/VehicleAdvertisementWrapper.java
package com.autoscout24.carfinder.ui;

import com.autoscout24.carfinder.arch.data.database.VehicleEntry;

public class VehicleAdvertisementWrapper {

    private VehicleEntry mVehicleEntry;
    //Could change this later to fetch custom Advertisements
    private String mAdvertisement;

    public VehicleAdvertisementWrapper(VehicleEntry vehicleEntry) {
        this.mVehicleEntry = vehicleEntry;
    }

    public VehicleAdvertisementWrapper(String advertisement) {
        this.mAdvertisement = advertisement;
    }

    public VehicleEntry getVehicleEntry() {
        return mVehicleEntry;
    }

    public String getAdvertisement() {
        return mAdvertisement;
    }

    public boolean isVehicle() {
        return mVehicleEntry != null ? true : false;
    }

    public boolean isAdvertisement() {
        return mAdvertisement != null ? true : false;
    }

    @Override
    public String toString() {
        return mVehicleEntry != null ? "Vehicle" : "Advertisement";
    }
}

File: app/src/main/java/com/autoscout24/carfinder/ui/VehicleAdvertisementWrapper.kt
package com.autoscout24.carfinder.ui

import com.autoscout24.carfinder.arch.data.database.VehicleEntry

class VehicleAdvertisementWrapper {

    var vehicleEntry: VehicleEntry? = null
    //Could change this later to fetch custom Advertisements
    var advertisement: String? = null

    fun isVehicle() = vehicleEntry != null
    fun isAdvertisement() = advertisement != null

    override fun toString(): String {
        return if (vehicleEntry != null) "Vehicle" else "Advertisement"
    }

    constructor(vehicleEntry: VehicleEntry) {
        this.vehicleEntry = vehicleEntry
    }

    constructor(advertisement: String) {
        this.advertisement = advertisement
    }
}